d2graph/d2graph.go

```diff
@@ -898,17 +898,36 @@ func appendTextDedup(texts []*d2target.MText, t *d2target.MText) []*d2target.MTe
 func (g *Graph) SetDimensions(mtexts []*d2target.MText, ruler *textmeasure.Ruler, fontFamily *d2fonts.FontFamily) error {
 	for _, obj := range g.Objects {
 		obj.Box = &geo.Box{}
-		// TODO fix edge cases for unnamed class etc
-		// Image shapes can set their own widths/heights
-		if obj.Attributes.Label.Value == "" && obj.Attributes.Shape.Value != d2target.ShapeImage {
-			obj.Width = 100
-			obj.Height = 100
-			continue
+
+		var setWidth int
+		var setHeight int
+		if obj.Attributes.Width != nil {
```

Comment:
can't it be an empty string here?

Reply:
the compiler already validates before this so it is a valid width